packages/block-editor/src/components/color-palette/with-color-context.js

```diff
@@ -23,7 +18,8 @@ export default createHigherOrderComponent( ( WrappedComponent ) => {
 			props.disableCustomColors === undefined
 				? disableCustomColorsFeature
 				: props.disableCustomColors;
-		const hasColorsToChoose = ! isEmpty( colors ) || ! disableCustomColors;
+		const hasColorsToChoose =
+			( colors && colors.length > 0 ) || ! disableCustomColors;
```

Comment:
The `colors` argument is expected to be an array:

https://github.com/WordPress/gutenberg/blob/8ff2348c3c40cf46e18c5492df3f2d57e28fa82b/packages/block-editor/src/components/color-palette/test/control.js#L19